src/test/java/com/puppycrawl/tools/checkstyle/xpath/XpathMapperTest.java

```diff
@@ -786,6 +786,169 @@ public void testFourthCaseGroup() throws Exception {
         assertArrayEquals("Result nodes differ from expected", expected, actual);
     }
 
+    @Test
+    public void testQueryElementFollowingSibling() throws Exception {
+        final String xpath = "//METHOD_DEF/following-sibling::*";
+        final RootNode rootNode = getRootNode("InputXpathMapperAst.java");
+        final DetailAST[] actual = convertToArray(getXpathItems(xpath, rootNode));
+        final DetailAST expectedMethodDefNode = getSiblingByType(rootNode.getUnderlyingNode(),
+                TokenTypes.CLASS_DEF)
+                .findFirstToken(TokenTypes.OBJBLOCK)
+                .findFirstToken(TokenTypes.METHOD_DEF)
+                .getNextSibling();
+        final DetailAST[] expected = {expectedMethodDefNode,
+                expectedMethodDefNode.getNextSibling()};
```

Comment:
it might be personal, but it took me some time to verify this, `expectedMethodDefNode.getNextSibling()` is very abstract. So human can not verify this test, without execution of it or deep dive in Input AST structure.
Does it make sense to extend test with `assertThat("...", RCURLY,  expectedMethodDefNode.getNextSibling().getType())`;
So users we see what is expected. Yes , flollowing `assertArrayEquals` will do this, but with extra assert test expected result could be verified by human.

Reply:
done